docs/source/topics/calculations/usage.rst

```diff
@@ -493,6 +493,37 @@ The parser implementation can then parse these files and store the relevant info
 After the parser terminates, the engine will take care to automatically clean up the sandbox folder with the temporarily retrieved files.
 The contract of the 'retrieve temporary list' is essentially that the files will be available during parsing and will be destroyed immediately afterwards.
 
+.. _topics:calculations:usage:calcjobs:stashing:
+
+Stashing files on the remote
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 1.6.0
+
+The `stash` option namespace allows a user to specify certain files that are created by the calculation job to be stashed somewhere on the remote.
+This can be useful if those files need to be stored for a longer time than the scratch space where the job was run is typically not cleaned for,
+but need to be kept on the remote machine and not retrieved.
+Examples are files that are necessary to restart a calculation but are too big to be retrieved and stored permanently in the local file repository.
+
```

Comment:
It would maybe be instructive to give first (or somewhere else) an example of the input structure. This visual hint may make it easier for a user to immediately form an image of how this works. E.g. something like:
```python
from aiida.common.datastructures import StashMode

inputs = {
    'code': ....,
    ...
    'metadata': {
        'options': {
            'stash': {
                'source_list': ['aiida.out', 'output.txt'],
                'target_base': '/storage/project/stash_folder',
                'stash_mode': StashMode.COPY.value,
            }
        }
    }
}
```

Reply:
ok added